### ISSUE TITLE
Duplicate blocks rule should ignore InitializeComponent in .Net views

### ISSUE THREAD

**Issue by movedoa** [Riverside-Software/sonar-openedge#99]
The user has no control over this code and it generates many false positives.
For example, the designer generates the following code:

```
DEFINE VARIABLE appearance1 AS Infragistics.Win.Appearance NO-UNDO.
appearance1 = NEW Infragistics.Win.Appearance().
@VisualDesigner.FormMember (NeedsInitialize="true":U).
DEFINE VARIABLE appearance2 AS Infragistics.Win.Appearance NO-UNDO.
appearance2 = NEW Infragistics.Win.Appearance().
@VisualDesigner.FormMember (NeedsInitialize="true":U).
DEFINE VARIABLE appearance3 AS Infragistics.Win.Appearance NO-UNDO.
appearance3 = NEW Infragistics.Win.Appearance().
@VisualDesigner.FormMember (NeedsInitialize="true":U).
DEFINE VARIABLE appearance4 AS Infragistics.Win.Appearance NO-UNDO.
appearance4 = NEW Infragistics.Win.Appearance().
@VisualDesigner.FormMember (NeedsInitialize="true":U).
```

**Comment by gquerret** [comment]
Is it possible to add an annotation to the method ? And does the annotation stay when file is saved in PDS ?

**Comment by movedoa** [comment]
Maybe but this would be a lot of work if you use .NET GUI for all your windows.
Can you check for comments?
The Method alway starts with the following comment which is generated by the PDS.

`METHOD PRIVATE VOID InitializeComponent(  ):

      /* NOTE: The following method is automatically generated.

      We strongly suggest that the contents of this method only be modified using the
      Visual Designer to avoid any incompatible modifications.

      Modifying the contents of this method using a code editor will invalidate any support for this file. */`

**Comment by gquerret** [comment]
It will be possible to select a set of method names to be excluded from CPD. Will be available in next version.

**Comment by gquerret** [comment]
Added two new properties (`sonar.oe.cpd.skip_methods` and `sonar.oe.cpd.skip_procedures`) in order to skip some methods and procedures/functions from the CPD engine.

However, I still think using annotations is the way to go ;-)

**Comment by gquerret** [comment]
Merged in master
